trl/models/modelling_sd_base.py

```diff
@@ -0,0 +1,405 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from diffusers import DDIMScheduler, StableDiffusionPipeline
+from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion import rescale_noise_cfg
```

Comment:
these imports needs to be protected with `is_diffusers_available()` similarly as PEFT: https://github.com/lvwerra/trl/blob/main/trl/import_utils.py#L24

Reply:
I have another question for you: 

I am not sure if this is the `below` comment you mentioned here

> After thinking a bit, we should maybe protect that whole import with is_diffusers_available() instead of my comment below, what do you think?

If not, my question: 
since the `is_diffusers_available` is responsible for conditionally importing all the functions related to DDPO at the __init__ level (at the main `__init__`, the trainer directory level `__init__` and models directory level `__init__`), would guarding it here be an overkill? I guess I'm not able to reason why we would want it here, if you could, please educate me on the use-case

Reply:
We do it this way to keep the total dependency of the library lightweight. So, if a user doesn't need diffusion related utilities, they don't have to `diffusers`.

Reply:
> We do it this way to keep the total dependency of the library lightweight
I understand this portion. I guess I'm asking if it's the guard is necessary at this file level if I've already put in guard at the __init__ levels?